great_expectations/data_context/data_context.py

```diff
@@ -2348,17 +2341,16 @@ def validations_store_name(self):
     def validations_store(self):
         return self.stores[self.validations_store_name]
 
-    def _compile_evaluation_parameter_dependencies(self, expectation_suite):
+    def _compile_evaluation_parameter_dependencies(self):
         self._evaluation_parameter_dependencies = {}
-
-        # only if we don't have an expectation suite do we do a key scan
-        if expectation_suite is None:
-            for key in self.expectations_store.list_keys():
+        for key in self.expectations_store.list_keys():
+            try:
                 expectation_suite = self.expectations_store.get(key)
                 if not expectation_suite:
                     continue
+            except InvalidKeyError:
+                continue
```

Comment:
Is there a reason we use `try/except` here? I believe the underlying `get()` call will return `None` if it doesn't have the key.

I believe we can get rid of L2347 and 2351-2352 in order to revert the changes made in https://github.com/great-expectations/great_expectations/pull/3377

Reply:
this was added (try/except on top of the if not) because of this particular test, test_tuple_s3_store_backend_expectation_suite_and_validation_operator_share_prefix it produced this Error (InvalidKeyError), thought it was intended maybe smth is broken in the s3 backend?

Reply:
Hmm that's a good point! I was just wary because I don't believe we had that `try/except` before.

Let me dig into the code and see if I can't figure out what's going on.

Reply:
Awesome, lemme know if I can be of use in any way

Reply:
so it appears that the specific test we're failing here is one that was added in the PR we're trying to revert. I think it makes sense to comment out of that test and add a small note about why that's an appropriate measure here.

would you mind if I committed to your branch? I could provide a bit of context and then I think we're good to merge here!

Reply:
Oh I see that makes sense now.
Sure thing, feel free to do so 😄